src/vanta_ledger/main.py

```
@@ -0,0 +1,270 @@
+    
+    Raises:
+        HTTPException: If the MongoDB connection or operation fails.
+    """
+    try:
+        client = get_mongo_client()
+        db = client.vanta_ledger
+        result = db.test.insert_one({"test": "data", "timestamp": datetime.now()})
+        db.test.delete_one({"_id": result.inserted_id})
+        return {"status": "MongoDB connection successful"}
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"MongoDB connection failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)